[WIP] Allow org admins to remove labels

rooftopcellist opened the pull request:
##### SUMMARY

Org Admin's couldn't unattach a label from a WFJT because the `can_unattach()` method in it's access class didn't cover that case.  

##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - Feature Pull Request

##### COMPONENT NAME
<!--- Name of the module/plugin/module/task -->
 - API


##### AWX VERSION
<!--- Paste verbatim output from `make VERSION` between quotes below -->
```
11.2.0
```


##### ADDITIONAL INFORMATION


I would like to add a test for this.  I am also still investigating why this is not needed for Job Templates.  There may be a better place for this fix.

rooftopcellist committed:
allow org admins to remove labels

AlanCoding commented:
why is this not needed for attaching, in addition to JTs? also curious about that.

jbradberry commented:
:eyes:

jbradberry reviewed (changes requested):
Based on my investigation just now on the devel branch, the issue is that a `TypeError` is raised on line 498 due to a typo.  Fixing the typo appears to allow an org admin to remove the label without problem, unless there is more to the issue than I realize.

https://github.com/ansible/awx/blob/f05bd0568fb65a09fd2a83639578b70f799c131e/awx/main/access.py#L498

rooftopcellist commented:
@jbradberry is absolutely right.  I just updated the PR.  

@AlanCoding It is not needed for attaching, I just reverted my prior change to the unattach method.  

It looks like this was just a typo in this commit: https://github.com/ansible/awx/commit/4eab362318a706290ae005c2e5c2840605735962#diff-c14c9d695bfbeb70889f2b42b18ed42dR495

jbradberry reviewed (approved):
:moyai: :+1:
